spec/controllers/api/v2/poly_metadata_controller_spec.rb

```diff
@@ -0,0 +1,98 @@
+            }
+          }
+        ]
+      }
+    end
+
+    describe 'with valid payload' do
+      before do
+        post '/api/v2/poly_metadata/bulk_create', params: valid_payload, as: :json
+      end
+
+      it 'creates the expected number of records' do
+        expect(PolyMetadatum.count).to eq(2)
+      end
+
+      it 'returns status created' do
+        expect(response).to have_http_status(:created)
+      end
+
+      it 'returns the created poly_metadata records' do
+        body = response.parsed_body
+        ids = PolyMetadatum.pluck(:id).map(&:to_s)
+
+        expect(body['data']).to contain_exactly({
+                                                  'id' => ids.first,
+                                                  'type' => 'poly_metadata',
+                                                  'attributes' => { 'key' => 'key_1', 'value' => 'value_1' },
+                                                  'relationships' => {
+                                                    'metadatable' => {
+                                                      'data' => { 'type' => 'aliquot', 'id' => aliquot.id.to_s }
+                                                    }
+                                                  }
+                                                }, {
+                                                  'id' => ids.second,
+                                                  'type' => 'poly_metadata',
+                                                  'attributes' => { 'key' => 'key_2', 'value' => 'value_2' },
+                                                  'relationships' => {
+                                                    'metadatable' => {
+                                                      'data' => { 'type' => 'aliquot', 'id' => aliquot.id.to_s }
+                                                    }
+                                                  }
+                                                })
+      end
+    end
+
+    describe 'with invalid payload' do
+      let(:bad_payload) do
+        valid_payload.deep_dup.tap do |payload|
+          payload[:data] << {
+            type: 'poly_metadata',
+            attributes: { key: nil, value: 'missing key' }, # invalid
+            relationships: {
+              metadatable: { data: { type: 'aliquot', id: aliquot.id.to_s } }
+            }
+          }
+        end
+      end
+
+      it 'does not create any records' do
+        expect do
+          suppress(Exception) do
+            post '/api/v2/poly_metadata/bulk_create', params: bad_payload, as: :json
+          end
+        end.not_to change(PolyMetadatum, :count)
+      end
+
+      it 'raises a validation error' do
+        expect do
+          post '/api/v2/poly_metadata/bulk_create',
+               params: bad_payload,
+               as: :json
+        end.to raise_error(ActiveRecord::RecordInvalid, /Key can't be blank/)
```

Comment:
This will cause a 500 error and the client will not be able to parse the result. 

Since this is a custom controller method, the result can be json formatted to approximate https://jsonapi.org/format/#error-objects . For example, with details field as e.record.errors.full_messages where e is the caught exception, title field as 'Validation failed' . 

We have a similar approaches implemented in different ways, which may be refactored.
- preview custom controller method for bait_library_layouts resource.
- register_stock_for_plate method for plates resource